src/rust/engine/sharded_lmdb/src/lib.rs

```diff
@@ -327,25 +327,70 @@ impl ShardedLmdb {
       .await
   }
 
+  ///
+  /// Singular form of `Self::exists_batch`. When checking the existence of more than one item,
+  /// prefer `Self::exists_batch`.
+  ///
   pub async fn exists(&self, fingerprint: Fingerprint) -> Result<bool, String> {
+    let missing = self.get_missing_fingerprints(vec![fingerprint]).await?;
+    Ok(!missing.contains(&fingerprint))
+  }
+
+  ///
+  /// Determine which of the given Fingerprints are not already present in the store.
+  ///
+  pub async fn get_missing_fingerprints(
```

Comment:
🤔 might be cleaner overall if I have this return present (vs. missing) fingerprints, and adjust the checks in the layer above accordingly